src/app/client/src/app/modules/notes/components/note-list/note-list.component.ts

```diff
@@ -0,0 +1,247 @@
+import { ResourceService, ToasterService, FilterPipe, ServerResponse, RouterNavigationService } from '@sunbird/shared';
+import { NotesService } from '../../services';
+import { UserService, ContentService } from '@sunbird/core';
+import { Component, OnInit, Pipe, PipeTransform, Input } from '@angular/core';
+// import { NoteFormComponent } from '../note-form/note-form.component';
+import { DatePipe } from '@angular/common';
+import { ActivatedRoute, Router } from '@angular/router';
+import { SuiModal, ComponentModalConfig, ModalSize, SuiModalService } from 'ng2-semantic-ui';
+import { INotesListData, ICourseDetails } from '@sunbird/notes';
+/**
+ * This component contains 2 sub components
+ * 1)NoteForm: Provides an editor to create and update notes.
+ * 2)DeleteNote: To delete an existing note.
+ */
+
+@Component({
+  selector: 'app-note-list',
+  templateUrl: './note-list.component.html',
+  styleUrls: ['./note-list.component.css']
+  // entryComponents: [NoteFormComponent]
+})
+export class NoteListComponent implements OnInit {
+
+  @Input() courseDetails: ICourseDetails;
+  /**
+   * This variable helps in displaying and hiding page loader.
+   * By default it is assigned a value of 'true'. This ensures that
+   * the page loader is displayed the first time the page is loaded.
+   */
+  showLoader = true;
+
+  /**
+   * The 'sortOrder' variable helps in making sure that the array of notes
+   * retrieved while making the search API call is sorted in descending order.
+   */
+  sortOrder = 'desc';
+
+  /**
+   * The notesList array holds the entire list of existing notes. Each
+   * note is saved as an object.
+   */
+  notesList: Array<INotesListData> = [];
```

Comment:
Rename INotesListData to NoteData as it represents data for a single note not a list of notes.